UnicornRender/video/source/Mesh.cpp

```diff
@@ -23,27 +28,33 @@ void Mesh::SetMeshData(const std::vector<Vertex>& vertices, const std::vector<ui
     VerticesUpdated.emit();
 }
 
-void Mesh::SetMaterial(Material material)
+void Mesh::SetMaterial(std::shared_ptr<Material> material)
 {
     m_material = material;
+    m_material->DataUpdated.connect(this, &Mesh::OnMaterialUpdated);
 
-    MaterialUpdated.emit();
+    this->MaterialUpdated.emit();
```

Comment:
I think it would also make sense to pass the material just in case listeners need it (it seems logical to pass values to `Updated` signals so that they're not requesting them on their own and if they only need this as a trigger, they can ignore the value

Reply:
Also why `this->`? Also why not just call `OnMaterialUpdated()`?

Reply:
Can't get what you want. Srrry